test/e2e/v4/plugin_cluster_test.go

```diff
@@ -219,9 +219,8 @@ func Run(kbc *utils.TestContext, hasWebhook, isToUseInstaller, isToUseHelmChart,
 
 		if hasMetrics {
 			By("labeling the namespace to allow consume the metrics")
-			_, err = kbc.Kubectl.Command("label", "namespaces", kbc.Kubectl.Namespace,
-				"metrics=enabled")
-			ExpectWithOffset(2, err).NotTo(HaveOccurred())
+			Expect(kbc.Kubectl.Command("label", "namespaces", kbc.Kubectl.Namespace,
```

Comment:
I started to work through it a bit - and initially changed a couple of more things (also incorrect ones). So I didn't want to move everything in one go. I then focused really on all the "...WithOffset" calls and the functions passed to Eventually(...)

Happy to add also change it in more places. I agree that it's better. Maybe just as a follow up pr? Or is there a preference to have it more in one PR?

Reply:
It's probably ok to do it in several PRs.  I'd keep it focused and to-the-point, so that PRs don't need to live too long and need constant rebasing etc.  The PR you have already fixes a lot of actual bugs (i.e. the Eventually() calls that use the global assertions).

I remember [a few months back, when I fixed assertion code like this](https://github.com/kubernetes-sigs/kubebuilder/pull/4197/commits/618a87b1e2e9b6b99ff6d2893fd9ed9207cd7b56#diff-efb8aa989aa88823884ccfe4e45f2dde83ee931d9d0b4cec73e5d42c105fb82aL133), I discovered that the test itself was [testing the completely wrong thing](https://github.com/mogsie/kubebuilder/blob/3352876ff7466ac2c5ffc636a9bb9bc2808b3d5c/testdata/project-v4-multigroup/internal/controller/example.com/busybox_controller_test.go#L132), and the incorrect assertions just happened to allow the test to pass.

So, when you do fix things like this, you might end up with tests that don't pass, and it might just be that the test itself was not testing what the original authors thought it was testing.